### ISSUE TITLE
Entry queries without section argument are scoped to author, regardless of settings

### ISSUE THREAD

**Issue by chrishale** [jamesedmonston/graphql-authentication#50]
I'm having an issue when performing queries without a section argument (I quite often already have an ID or URI from another query). For example, if I query the following with a valid JWT - no entry is returned. But without the JWT, the query succeeds as it should.
`
query {
  entry(uri: "about") {
    id
  }
}
`
For reference, I'm not currently using the "Restricted Entry Queries" sections for my schema, but that doesn't appear to make any change to the logic.

Looking at the Entry resolver file: https://github.com/jamesedmonston/graphql-authentication/blob/104065efcef403486d4b5a0bd963f7fdce79cde8/src/resolvers/Entry.php#L84

It looks as though the else statement above is choosing to scope the query to the owner of the JWT when the section argument isn't set. 

Am I doing something wrong here, is this a limitation for some other logic or a bug?

Thanks!

**Comment by jamesedmonston** [comment]
@chrishale Thanks for pointing this out. I've been meaning to add a note to the docs about this as it isn't the first time it's come up.

The automatic injection of the `authorId` argument on queries that don't specify a `section`/`sectionId` argument is a security fallback for sites that _do_ use the restricted query settings. It also helps to keep limit/pagination working correctly.

In the following query, if the plugin were to grab the entries and _then_ filter out private entries, it could potentially return less than the requested amount:

```
query Entries {
  entries(limit: 5) {
    id
    title
  }
}
```

tl;dr: the plugin requires it so it can check against the schema and restriction settings at query time, rather than needing to filter out potentially private entries after they've been retrieved.

Are you able to add the `section` argument to your queries, or is this going to cause you issues?

I'll make a note to get this added to the docs 👍🏻

Cheers

**Comment by chrishale** [comment]
@jamesedmonston Thanks for your reply, this makes sense now. Perhaps there is room for an `else if` in there for when no sections are selected in the "Restricted Entry Queries" settings, because in that instance - it shouldn't need to set the authorId?

This won't necessarily fix my use-case as I will need to take advantage of the Restricted Entry Queries at some point, so probably not an urgent addition. I'll work on trying to pass in the section argument, or perhaps just omit the JWT for those queries where it isn't actually needed. Thanks again!

**Comment by jamesedmonston** [comment]
@chrishale I've _finally_ added a note about this to the docs! https://graphql-authentication.jamesedmonston.co.uk/extra/fy
